File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int sum = 0;
        int head_pos = 0;
        int min_len = nums.length + 69;
        
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            while (sum >= target) {
                if (min_len > i + 1 - head_pos) {
                    min_len = i + 1 - head_pos;
                }
                sum -= nums[head_pos];
                head_pos++;
            }
        }
        
        return (min_len == nums.length + 69) ? (0) : (min_len);
    }
}

File: top-interview-150/0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        sum, head_pos, min_len = 0, 0, len(nums) + 69
        
        for i in range(len(nums)):
            sum += nums[i]
            while sum >= target:
                if min_len > i + 1 - head_pos:
                    min_len = i + 1 - head_pos
                sum -= nums[head_pos]
                head_pos += 1
                
        return 0 if min_len == len(nums) + 69 else min_len